builtin/providers/aws/data_source_aws_sns_test.go

```diff
@@ -0,0 +1,45 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccAwsSnsTopicDataSource_noMatchReturnsError(t *testing.T) {
+	name := "hashicorp.com"
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccCheckAwsSnsTopicDataSourceConfig(name),
```

Comment:
I'd love to see a test that creates an SNS Topic and then shows the data_source fetching the ARN